src/main/main.js

```diff
@@ -163,13 +162,19 @@ function initializeArgs() {
   }
 }
 
-function initializeConfig() {
-  //registryConfig = new RegistryConfig();
-  config = new Config(app.getPath('userData') + '/config.json');
-  config.on('update', handleConfigUpdate);
-  config.on('synchronize', handleConfigSynchronize);
-  config.on('darkModeChange', handleDarkModeChange);
-  WindowManager.setConfig(config.data, config.showTrayIcon, deeplinkingUrl);
+async function initializeConfig() {
+  const loadConfig = new Promise((resolve, reject) => {
+    config = new Config(app.getPath('userData') + '/config.json');
+    config.once('update', (configData) => {
+      config.on('update', handleConfigUpdate);
+      config.on('synchronize', handleConfigSynchronize);
+      config.on('darkModeChange', handleDarkModeChange);
+      handleConfigUpdate(configData);
+      resolve();
```

Comment:
in windows, there is a bigger delay until the registry is read, and this causes the config to not be loaded in time, making other parts fail

Reply:
Do we need the second `await` in this function? Couldn't we just return the promise and the `await` in `initialize()` will take care of it?

Reply:
config is being used by `initializeBeforeAppReady` so we have to wait for it to finish. In windows it needs to go to the registry and get the info from there and it ends up being a race condition, so it feel safer this way.